auth/auth.go

```diff
@@ -59,7 +59,7 @@
 //
 // First create an Auth object:
 //
-//  myAuth := auth.New(tcclient.Credentials{ClientId: "myClientId", AccessToken: "myAccessToken"})
+//  myAuth := auth.New(&tcclient.Credentials{ClientId: "myClientId", AccessToken: "myAccessToken"})
```

Comment:
Mostly for consistency with other APIs, but yes it means moving around and duplicating fewer bits. Another reason is to simplify testing if credentials have been initialised - instead of checking whether all the fields of a struct have the zero value, you can see if the pointer is nil or not. This is particularly useful when unmarshaling/marshaling to/from json representation, since you can distinguish between a non-existing value, and a zero value. Long story short, I've moved to the general principle of using pointers by default, unless there is a strong reason not to. This might also be influenced by my evil java past (where essentially all non-primitive types are backed by pointers).